### DISCUSSION TITLE
How to use generic type on a route handler?

### DISCUSSION THREAD

**Discussion by melokki** [actix/actix-web#2984] [Q&A] [answered]
Hello everyone

I am trying to use a generic type I am hitting some blocks.
I managed to fix most of them, but now I got one that I don't know how to fix

I am trying to create a route, but I am having the following issue

here is my code

```rust
.route(
                "/send",
                web::post().to(mails::app::controllers::mails_controller::store::<T>),
            )
```

for this line of code I am getting the following error

```rust
cannot find type `T` in this scope
not found in this scope
```

and I don't know how to fix this anymore

Any suggestions on what to do next?

**Comment by robjtede** [comment, marked as answer]
Generic handlers are fine but using a type parameter in that position requires a concrete type in Rust.

**Reply by melokki** [reply]
Thank you very much for your reply. Yes, I dropped this idea. 
managed to solve my problem using Enums  instead of generics, works like a charm :D :)
